toolkit/themes/shared/design-system/config.js

```diff
@@ -1,11 +1,101 @@
-
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 /* eslint-env node */
 
 const StyleDictionary = require("style-dictionary");
+const { formattedVariables } = StyleDictionary.formatHelpers;
+
+const MEDIA_QUERY_PROPERTY_MAP = {
+  "forced-colors": "forcedColors",
+  "prefers-contrast": "prefersContrast",
+};
+
+/**
+ * Formats built CSS to include "prefers-contrast" and "forced-colors" media
+ * queries.
+ *
+ * @param {object} args
+ *  Formatter arguments provided by style-dictionary. See more at
+ *  https://amzn.github.io/style-dictionary/#/formats?id=formatter
+ * @returns {string}
+ *  Formatted CSS including media queries.
+ */
+function hcmFormatter(args) {
+  return (
+    formatTokens({ args }) +
+    formatTokens({ mediaQuery: "prefers-contrast", args }) +
+    formatTokens({ mediaQuery: "forced-colors", args })
+  );
+}
+
+/**
+ * Formats a subset of tokens into CSS. Wraps token CSS in a media query when
+ * applicable.
+ *
+ * @param {object} tokenArgs
+ * @param {string} [tokenArgs.mediaQuery]
+ *  Media query formatted CSS should be wrapped in. This is used
+ *  to determine what property we are parsing from the token values.
+ * @param {object} tokenArgs.args
+ *  Formatter arguments provided by style-dictionary. See more at
+ *  https://amzn.github.io/style-dictionary/#/formats?id=formatter
+ * @returns {string} Tokens formatted into a CSS string.
+ */
+function formatTokens({ mediaQuery, args }) {
+  let prop = MEDIA_QUERY_PROPERTY_MAP[mediaQuery] ?? "value";
+  let dictionary = Object.assign({}, args.dictionary);
+  let tokens = [];
+
+  dictionary.allTokens.forEach(token => {
+    let value = token[prop] || token.original.value[prop]
+    if (value && typeof value !== "object") {
+      let formattedToken = transformTokenValue(token, prop, dictionary);
+      tokens.push(formattedToken);
+    }
+  });
+
+  dictionary.allTokens = dictionary.allProperties = tokens;
+
+  let formattedVars = formattedVariables({
+    format: "css",
+    dictionary,
+    outputReferences: args.options.outputReferences,
+  });
+
+  if (mediaQuery) {
+    return `
+@media (${mediaQuery}) {
+  :root {
+  ${formattedVars.split("\n").join(`\n  `)}
```

Comment:
oh no, my perf

I think we can avoid the split/join

```suggestion
${formattedVars}
```

Reply:
Unfortunately we seem to need this to get the right spacing for media query blocks ☹️ This is how the output looks for me if I just use `formattedVars` directly: 

<img width="398" alt="Screenshot 2024-01-29 at 10 00 36 AM" src="https://github.com/FirefoxUX/gecko-dev/assets/15138046/02b1b6f5-c754-4332-862d-e7cfae469770">

There could very well be a better way to do this + if we want to rely on something like ESLint for proper formatting down the line then we can definitely stop doing the formatting manually.